libminifi/src/sitetosite/RawSocketProtocol.cpp

```diff
@@ -395,97 +395,37 @@ bool RawSiteToSiteClient::getPeerList(std::vector<PeerStatus> &peers) {
   }
 }
 
-int RawSiteToSiteClient::writeRequestType(RequestType type) {
-  if (type >= MAX_REQUEST_TYPE)
-    return -1;
-
-  return peer_->writeUTF(SiteToSiteRequest::RequestTypeStr[type]);
-}
-
-int RawSiteToSiteClient::readRequestType(RequestType &type) {
-  std::string requestTypeStr;
-
-  int ret = peer_->readUTF(requestTypeStr);
-
-  if (ret <= 0)
-    return ret;
+  int RawSiteToSiteClient::writeRequestType(RequestType type) {
+    if (type >= MAX_REQUEST_TYPE)
+      return -1;
 
-  for (int i = NEGOTIATE_FLOWFILE_CODEC; i <= SHUTDOWN; i++) {
-    if (SiteToSiteRequest::RequestTypeStr[i] == requestTypeStr) {
-      type = (RequestType) i;
-      return ret;
-    }
+    return peer_->writeUTF(SiteToSiteRequest::RequestTypeStr[type]);
   }
 
-  return -1;
-}
-
-int RawSiteToSiteClient::readRespond(const std::shared_ptr<Transaction> &transaction, RespondCode &code, std::string &message) {
-  uint8_t firstByte;
```

Comment:
This is again the case where the base class doesn't need to deal with the the innards of reading bytes. Probably means they can be removed from the base and made pure virtual since I believe they are overridden in http site to site if I'm not mistaken.

Reply:
This isn't valid any longer. I think some parent class details changed, but I will have to further investigate the HTTP portion. I want to fully understand this. If it is a bug that may push when this should be merged, but once I take a closer look at HTTP it may become obvious. What the parent currently does, is an invalid reliance from the end of HTTP, but since things changed in some of the accessory code it may never get there, meaning we can make it pure virtual; however, I'll take a closer look at it while running to validate that it's not a bug before going that route.

Reply:
I'll also take another look at this PR closely and approve for merge in 0.7.0 if it doesn't change status quo. Thanks.

Reply:
I didn't see a response for this. I have some time in the next few days to run tests. What have you been able to run? Thanks!

Reply:
Two things I did:
-Execute site2site related unit tests
-Verified transfers from MiNiFi to NiFi using the C examples (which still rely on C++ S2S implementation atm)

Reply:
Yep, I usually do full scope: docker tests, site to site secure, unsecure, transfers -- in this case I'll do raw and http just to be safe ( in minifi note C agent ). -- once all that is done I'll take another look at the PR and approve for merge in 0.7.0. Thanks.